src/spikeinterface/core/sparsity.py

```diff
@@ -119,6 +125,87 @@ def unit_id_to_channel_indices(self):
                 self._unit_id_to_channel_indices[unit_id] = channel_inds
         return self._unit_id_to_channel_indices
 
+    def sparsify_waveforms(self, waveforms: np.ndarray, unit_index: int) -> np.ndarray:
+        """
+        Sparsify the waveforms according to a unit's channel sparsity.
+
+        The units are indexed by their index in the unit_ids list.
+
+        Given a unit index, this method selects only the active channels for
+        that unit and removes the rest.
+
+        Parameters
+        ----------
+        waveforms : np.array
+            Dense waveforms with shape (num_units, num_samples, num_channels).
+        unit_index : int
+            The index of the unit for which to sparsify the waveform.
+
+        Returns
+        -------
+        sparsified_waveforms : np.array
+            Sparse waveforms with shape (num_units, num_samples, num_active_channels).
+        """
+
+        assert_msg = (
+            "Waveforms must be dense to sparsify them. "
+            f"Their last dimension {waveforms.shape[-1]} must be equal to the number of channels {self.num_channels}"
+        )
+        assert self.are_waveforms_dense(waveforms=waveforms), assert_msg
+
+        unit_id = self.unit_ids[unit_index]
+        non_zero_indices = self.unit_id_to_channel_indices[unit_id]
+        num_sparse_channels = len(non_zero_indices)
+        sparsified_shape = waveforms.shape[:-1] + (num_sparse_channels,)
+        sparsified_waveforms = np.zeros(sparsified_shape, dtype=waveforms.dtype)
+        sparsified_waveforms[...] = waveforms[..., non_zero_indices]
```

Comment:
```suggestion
        sparsified_waveforms = waveforms[:, :, non_zero_indices]
```
should be the same, right?

Reply:
It should. And you mentioned it before.

Reply:
Got you now, I Have changed this. Thanks.